irohad/ordering/impl/on_demand_ordering_service_impl.cpp

```diff
@@ -61,11 +65,15 @@ void OnDemandOrderingServiceImpl::onBatches(consensus::Round round,
              round.block_round,
              round.reject_round);
 
+  auto unprocessed_batches =
+      boost::adaptors::filter(batches, [this](const auto &batch) {
+        return not this->batchAlreadyProcessed(*batch);
+      });
```

Comment:
The problem here is that we are working with ranges, which are lazy and, in case of filter, single pass.

This means that we cannot get size of the range without invoking it. This means that if we add size calculation, range will be invoked twice, which is not good.